libsolidity/analysis/PostTypeChecker.cpp

```diff
@@ -336,27 +342,38 @@ struct EventOutsideEmitErrorOutsideRevertChecker: public PostTypeChecker::Checke
 		if (*_functionCall.annotation().kind == FunctionCallKind::FunctionCall)
 			if (auto const* functionType = dynamic_cast<FunctionType const*>(_functionCall.expression().annotation().type))
 			{
+				if (!m_inRequire && functionType->kind() == FunctionType::Kind::Require)
+					m_inRequire = true;
```

Comment:
This should really an assert since this mechanism is not prepared to handle nested `require` even if it was allowed. It would set `m_inRequire` when entering the outer `require` but then incorrectly unset after exiting the inner one..

```suggestion

				if (functionType->kind() == FunctionType::Kind::Require)
				{
					solAssert(!m_inRequire);
					m_inRequire = true;
				}
```

And ` endVisit(FunctionCall)` needs a similar assert.